test/functional/apm_tests.js

```diff
@@ -125,10 +132,12 @@ describe('APM', function() {
           .then(function(r) {
             expect(r.insertedCount).to.equal(1);
 
-            var listener = require('../..').instrument(function(err) {
+            self.listener = require('../..').instrument(function(err) {
```

Comment:
I would prefer not to use the magic context for this.  I tend to create a global variable e.g. `test` or `testContext` to make it very explicit where all this stuff lives